dspace-api/src/main/java/org/dspace/administer/CreateAdministrator.java

```diff
@@ -64,24 +66,115 @@ public static void main(String[] argv)
+        options.addOption("h", "help", false, "explain create-administrator options");
         options.addOption("l", "last", true, "administrator last name");
         options.addOption("c", "language", true, "administrator language");
         options.addOption("p", "password", true, "administrator password");
 
-        CommandLine line = parser.parse(options, argv);
+        CommandLine line = null;
+
+        try {
+
+            line = parser.parse(options, argv);
+
+        } catch (Exception e) {
+
+            System.out.println(e.getMessage() + "\nTry \"dspace create-administrator -h\" to print help information.");
+            System.exit(1);
+
+        }
 
         if (line.hasOption("e") && line.hasOption("f") && line.hasOption("l") &&
             line.hasOption("c") && line.hasOption("p")) {
             ca.createAdministrator(line.getOptionValue("e"),
                                    line.getOptionValue("f"), line.getOptionValue("l"),
                                    line.getOptionValue("c"), line.getOptionValue("p"));
         } else {
-            ca.negotiateAdministratorDetails();
+
+            if (line.hasOption("h")) {
+                String header = "\nA command-line tool for creating an initial administrator for setting up a" +
+                    " DSpace site. Unless all the required parameters are passed it will" +
+                    " prompt for an e-mail address, last name, first name and password from" +
+                    " standard input.. An administrator group is then created and the data passed" +
+                    "  in used to create an e-person in that group.\n\n";
+                String footer = "\n";
+                HelpFormatter formatter = new HelpFormatter();
+                formatter.printHelp("dspace create-administrator", header, options, footer, true);
+                return;
+            } else {
+
+                ConfigurationService cfg = DSpaceServicesFactory.getInstance().getConfigurationService();
+
+                if (!line.hasOption('p') && line.hasOption("e")  && line.hasOption("f") && line.hasOption("l")
+                    && (line.hasOption("c") || (!line.hasOption("c")
+                    && cfg.getProperty("webui.supported.locales") == null))) {
+
+
+                    if (cfg.getProperty("webui.supported.locales") != null) {
+                        System.out.println("Select one of the following languages: "
+                            + cfg.getProperty("webui.supported.locales"));
+                        System.out.print("Language: ");
+                        System.out.flush();
+
+                        language = console.readLine();
+
+                        if (language != null) {
+                            language = language.trim();
+                            language = I18nUtil.getSupportedLocale(new Locale(language)).getLanguage();
+                        }
+                    }
+
+                    while (!dataOK) {
```

Comment:
This loop only allows the user to correct the password, which cannot be seen.  The original will ask for all of the user properties again.  Neither is what I would expect.

Suggestion:  factor out the password-prompting code into a new private method.  Loop there, not returning until both passwords match.  Return one of the character arrays after clearing the other one.  Call from `negotiateAdministratorDetails`.

Suggestion:  `negotiateAdministratorDetails` should display the previously-collected values (if any) when looping through its prompts.  An empty response to a prompt should keep the existing value.

Reply:
I am working on the changes. Will take sometime.

Reply:
I changed the code